File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/customerrole/CustomerRoleResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.customerrole;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import ru.herooo.mylanguagedb.entities.Color;
import ru.herooo.mylanguageweb.controllers.json.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.color.ColorResponseDTO;

public class CustomerRoleResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    private long id;

    @JsonProperty("title")
    private String title;

    @JsonProperty("path_to_image")
    private String pathToImage;

    @JsonProperty("code")
    private String code;

    @JsonProperty("message")
    private String message;

    @JsonProperty("color")
    private ColorResponseDTO color;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPathToImage() {
        return pathToImage;
    }

    public void setPathToImage(String pathToImage) {
        this.pathToImage = pathToImage;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public ColorResponseDTO getColor() {
        return color;
    }

    public void setColor(ColorResponseDTO color) {
        this.color = color;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/customer_role/response/customer_role_response_dto.js
import {
    ColorResponseDTO
} from "../../color/response/color_response_dto.js";

export class CustomerRoleResponseDTO {
    #id;
    #title;
    #pathToImage;
    #code;
    #description;
    #color;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#title = json["title"];
            this.#pathToImage = json["path_to_image"];
            this.#code = json["code"];
            this.#description = json["description"];

            let color = json["color"];
            if (color) {
                this.#color = new ColorResponseDTO(color);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getTitle() {
        return this.#title;
    }

    getPathToImage() {
        return this.#pathToImage;
    }

    getCode() {
        return this.#code;
    }

    getDescription() {
        return this.#description;
    }

    getColor() {
        return this.#color;
    }
}